packages/dashboard/src/components/dashboard/index.test.tsx

```diff
@@ -66,6 +66,47 @@ it('renders in readonly initially', function () {
   expect(queryByTestId(/time-selection/i)).toBeInTheDocument();
 });
 
+it('passes the correct viewMode to onSave', function () {
+  const onSave = jest.fn();
+
+  const savedConfig = {
+    displaySettings: {
+      numColumns: 100,
+      cellSize: 20,
+      numRows: 100,
+      significantDigits: 4,
+    },
+    widgets: [],
+    viewport: { duration: '5m' },
+  };
+
+  render(
+    <Dashboard
+      onSave={onSave}
+      dashboardConfiguration={savedConfig}
+      clientConfiguration={{
+        iotEventsClient: createMockIoTEventsSDK(),
+        iotSiteWiseClient:
+          createMockSiteWiseSDK() as unknown as IoTSiteWiseClient,
+        iotTwinMakerClient: {
+          send: jest.fn(),
+        } as unknown as IoTTwinMakerClient,
+      }}
+      initialViewMode='edit'
```

Comment:
What about `viewMode`?

Reply:
since i didn't do my original solution due to overrequesting data then i thought initialViewMode was still a more fitting name

the parent component does not keep track of the current viewMode, it just knows what the viewMode was onSave